### PR TITLE
Task/DES-1988: Add hazmapperMaps field to project metadata and display in project preview views.

### PR DESCRIPTION
**Pull request by duckonomy**
## Overview: ##
Adds the `hazmapperMaps` field to the Project models and displays it in the project views.
The fields are intended to be directly set/updated from Hazmapper (Will later be able to test with #911 and TACC-Cloud/hazmapper#70 [WIP]).

## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [DES-1988](https://jira.tacc.utexas.edu/browse/DES-1988)

## Summary of Changes: ##
Modified the Project-related models for django and elasticsearch to include the `hazmapperMaps` field, which is looks like (List of objects):
```
[{
  name: string (name of the map: given in hazmapper),
  path: string (),
  deployment: string (),
  uuid: string ()
}]
```

Also modified project views to include the field values.

## Testing Steps: ##
1. Create a maps in [hazmapper prod](https://hazmapper.tacc.utexas.edu/hazmapper/) and [hazmapper staging](https://hazmapper.tacc.utexas.edu/staging/)
2. Try making a request to the `api/projects/$UUID` either by adding a request value in the `update()` function of `manage-project.component.js`) or with postman/curl with #911 and the body :
```
{
    "uuid": $UUID,
    "hazmapperMaps": [
        {
            "name": $STAGING_MAP_NAME, 
            "path": $STAGING_MAP_PATH, 
            "uuid": $STAGING_MAP_UUID,
            "deployment": "staging"
        },
        {
            "name": $PROD_MAP_NAME, 
            "path": $PROD_MAP_NAME, 
            "uuid": $PROD_MAP_NAME,
            "deployment": "production"
        }
    ]
}
```
3. Ensure that the project-view of the project properly displays the list of hazmapper maps.
4. Ensure that the external link to hazmapper opens to the proper hazmapper map with the matching uuid and deployment.

## UI Photos:
Screenshot from 2021-11-09 17-47-05

## Notes: ##
Please let me know if I am missing something in regards to updating the project model's fields and project views.
There is also a lot of redundant code that I wasn't sure how to split at the moment (especially for the different project views). So, I just kept how it is now (similar to `onBrowse()`).

### CODE REVIEW

File: designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.js [hunk 1]
```diff
@@ -206,14 +206,37 @@ class CurationDirectoryCtrl {
...
-    else {
-        this.FileOperationService.openPreviewModal({api: 'agave', scheme: 'private', file})
+
+    filteredHazmapperMaps(maps) {
+        maps = maps ? maps : [];
+
+        maps.forEach(map => {
+            switch(map.deployment) {
+                case 'production':
+                    map.href = `https://hazmapper.tacc.utexas.edu/hazmapper/project/${map.uuid}`;
```
Comment (nathanfranklin) [inline designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.js:224]: what will we do about public maps?  I'm wondering if a user publishes a project, do the map owners need to make sure they allow public access (via make public map).  I'm not certain 🤔   We could just put some thoughts in a ticket for follow up later.

Comment (duckonomy) [reply designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.js:224]: @nathanfranklin Would it be best to have a separate case for that? Or could it be handled on the hazmapper side? Maybe by redirecting to the public map (if the user does not have access && there is a public map available)?

**Review by nathanfranklin — CHANGES_REQUESTED** [review]
Looks good; its gonna be great to have this connection between Hazmapper/DS for users to use 💯 

One comment for refactoring suggestion.

File: designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.html [hunk 2]
```diff
@@ -97,6 +97,17 @@
                 <td ng-if="$ctrl.browser.project.value.keywords">Keywords</td>
                 <td class="prj-data">{{ $ctrl.browser.project.value.keywords }}</td>
             </tr>
+            <tr ng-repeat="map in $ctrl.filteredHazmapperMaps($ctrl.browser.project.value.hazmapperMaps)">
```
[the remark this answers is not in the document]
Comment (duckonomy) [reply designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.html:100]: Gotcha, I'll try doing that.

[the remark this answers is not in the document]
Comment (duckonomy) [reply designsafe/static/scripts/data-depot/components/projects/curation-directory/curation-directory.component.html]: @nathanfranklin Resolved!

**Review by jarosenb — APPROVED** [review]
LGTM! Works with and without associated maps.

**Pull request merged**